src/core/MOM.F90

```diff
@@ -962,10 +962,12 @@ subroutine step_MOM(forces_in, fluxes_in, sfc_state, Time_start, time_int_in, CS
 
     !===========================================================================
     ! This is the second place where the diabatic processes and remapping could occur.
-    if (thermo_does_span_coupling .or. .not.do_dyn) then
```

Comment:
It now appears that there are cases where `do_diabatic` is not being initialized, but it is being used on line 972.  Logically the value of `do_diabatic` does not matter, but the use of an uninitialized variable will be flagged when certain debugging capabilities are enabled.  Please consider adding `else ; do_diabatic = .false.` just before line 971.